src/goto-symex/goto_symex_state.cpp

```diff
@@ -500,6 +493,32 @@ bool goto_symex_statet::l2_thread_read_encoding(
   return true;
 }
 
+/// Allocates a fresh L2 name for the given L1 identifier, and makes it the
+/// latest generation on this path.
+void goto_symex_statet::increase_generation(
+  const irep_idt l1_identifier,
+  const ssa_exprt &lhs)
+{
+  auto current_emplace_res =
+    level2.current_names.emplace(l1_identifier, std::make_pair(lhs, 0));
```

Comment:
It looks like this method belongs to symex_level2t. ￼￼:leaves:

Reply:
This was considered but not done because we didn't want `symex_level2t` concerned with the global name allocator (only with the path-local map)